src/envs/behavior_options.py

```diff
@@ -461,46 +462,160 @@ def create_grasp_policy(
+                ))
+                if len(plan) == 1:
+                    plan_executed_forwards = True
+
+            plan.pop(0)
+            return low_level_action, done_bit
+
+            ###
 
         elif (plan_executed_forwards and not tried_closing_gripper):
             # Close the gripper to see if you've gotten the
             # object
-            ret_action = np.zeros(env.action_space.shape, dtype=float)
-            ret_action[16] = 1.0
+            low_level_action = np.zeros(env.action_space.shape, dtype=float)
+            low_level_action[16] = 1.0
             tried_closing_gripper = True
+            plan = reversed_plan
+            return low_level_action, False
 
         else:
-            # Grasping Phase 3: getting the hand back to
+            expected_pos = np.array(plan[0][0:3])
+            expected_orn = np.array(plan[0][3:])
+            ###
+            # 2. if error is greater that MAX_ERROR
+            if not np.allclose(current_pos, expected_pos,
+                               atol=atol_xyz) or not np.allclose(
+                                   current_orn, expected_orn, atol=atol_theta):
+                # 2.a take a corrective action
+                if len(plan) <= 1:
+                    done_bit = True
+                    print("PRIMITIVE: grasp policy completed execution!")
+                    return np.zeros(env.action_space.shape,
+                                    dtype=np.float32), done_bit
+                low_level_action = (get_delta_low_level_hand_action(
+                    env.robots[0].parts["body"],
+                    np.array(current_pos),
+                    np.array(current_orn),
+                    np.array(plan[0][0:3]),
+                    np.array(plan[0][3:]),
+                ))
+
+                # But if the corrective action is 0
+                if np.allclose(
+                        low_level_action,
+                        np.zeros((env.action_space.shape, 1)),
+                        atol=atol_vel,
+                ):
+                    low_level_action = (get_delta_low_level_hand_action(
+                        env.robots[0].parts["body"],
+                        np.array(current_pos),
+                        np.array(current_orn),
+                        np.array(plan[1][0:3]),
+                        np.array(plan[1][3:]),
+                    ))
+                    plan.pop(0)
+
+                return low_level_action, False
+
+        if len(plan) == 1:  # In this case, we're at the final position
+            low_level_action = np.zeros(env.action_space.shape, dtype=float)
+            done_bit = True
+            print("PRIMITIVE: grasp policy completed execution!")
+
+        else:
+            # Placing Phase 3: getting the hand back to
             # resting position near the robot.
-            ret_action = get_delta_low_level_hand_action(
-                env,
+            low_level_action = get_delta_low_level_hand_action(
```

Comment:
add comment describing what's going on with all this indexing into reversed_plan?